src/EvoSC.Common/Services/PlayerCacheService.cs

```diff
@@ -0,0 +1,244 @@
+    {
+        get
+        {
+            lock (_onlinePlayersMutex)
+            {
+                return _onlinePlayers.Values;
+            }
+        }
+    }
+
+    public PlayerCacheService(IEventManager events, IServerClient server, ILogger<PlayerCacheService> logger, IPlayerRepository playerRepository)
+    {
+        _server = server;
+        _logger = logger;
+        _playerRepository = playerRepository;
+        
+        SetupEvents(events);
+    }
+
+    private void SetupEvents(IEventManager events)
+    {
+        events.Subscribe(s => s
+            .WithEvent(GbxRemoteEvent.PlayerConnect)
+            .WithInstance(this)
+            .WithInstanceClass<PlayerCacheService>()
+            .WithHandlerMethod<PlayerConnectEventArgs>(OnPlayerConnectAsync)
+            .WithPriority(EventPriority.High)
+            .AsAsync()
+        );
+        
+        events.Subscribe(s => s
+            .WithEvent(GbxRemoteEvent.PlayerDisconnect)
+            .WithInstance(this)
+            .WithInstanceClass<PlayerCacheService>()
+            .WithHandlerMethod<PlayerDisconnectEventArgs>(OnPlayerDisconnectAsync)
+            .WithPriority(EventPriority.High)
+            .AsAsync()
+        );
+        
+        events.Subscribe(s => s
+            .WithEvent(GbxRemoteEvent.PlayerInfoChanged)
+            .WithInstance(this)
+            .WithInstanceClass<PlayerCacheService>()
+            .WithHandlerMethod<PlayerInfoChangedEventArgs>(OnPlayerInfoChangedAsync)
+            .WithPriority(EventPriority.High)
+            .AsAsync()
+        );
+
+        _server.Remote.OnConnected += OnServerConnectedAsync;
+    }
+
+    private async Task OnServerConnectedAsync()
+    {
+        try
+        {
+            await UpdatePlayerListAsync();
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Failed to update player list cache");
+            throw;
+        }
+    }
+
+    private async Task OnPlayerInfoChangedAsync(object sender, PlayerInfoChangedEventArgs e)
+    {
+        var accountId = PlayerUtils.ConvertLoginToAccountId(e.PlayerInfo.Login);
+        await ForceUpdatePlayerAsync(accountId);
+    }
+
+    private Task OnPlayerDisconnectAsync(object sender, PlayerDisconnectEventArgs e)
+    {
+        var accountId = PlayerUtils.ConvertLoginToAccountId(e.Login);
+
+        lock (_onlinePlayersMutex)
+        {
+            if (_onlinePlayers.ContainsKey(accountId))
+            {
+                _onlinePlayers.Remove(accountId);
+            }
+        }
+        
+        return Task.CompletedTask;
+    }
+
+    private async Task OnPlayerConnectAsync(object sender, PlayerConnectEventArgs e)
+    {
+        try
+        {
+            var accountId = PlayerUtils.ConvertLoginToAccountId(e.Login);
+            await ForceUpdatePlayerAsync(accountId);
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Failed to cache player");
```

Comment:
Think it could be a good idea to actually log which player that can't be added to ease the debugging process if this were to happen regularly or irregularly.
```suggestion
            _logger.LogError(ex, "Failed to cache player with login {AccountId}", accountId);
```